backend/api/v1/endpoints/connections.py

```
@@ -56,13 +56,38 @@ def list_connections(db: Session = Depends(get_db), current_user: User = Depends
+            Connection.status == "pending"
+        )
+
+        # Get users who are not in any of the above lists
+        users = db.query(User).filter(
+            User.id.notin_(friends),
+            User.id.notin_(sent_requests),
+            User.id.notin_(received_requests),
+            User.id != current_user.id  # Exclude self
+        ).all()
+
         return users
```

Ruff (0.8.2)

Consider moving this statement to an `else` block

(TRY300)